### PR TITLE
Enh registration

### PR DESCRIPTION
**Pull request by ArthurCourselle**
Enhance the registration process:
  - The user can choose the radius of the circle where the cross-correlation is computed. A spinbox is added in the GUI.
  - The reference image is taken from the accumulation queue after the first accumulation at each pipe refresh.
  - The registration is now working in CLI mode.
  - The registration is now in the compute settings. The registration zone (circle radius) is saved in the compute settings.
- Refactorisation of fn_compute_vect:
  - Do not inherit std::vector anymore.
  - Associate each functions to a unique ID. This allows to track each functions and give more flexibility,
    for example: removing a function from the vector.
    
    Also merging the new ps1 for registration.

### CODE REVIEW

File: Holovibes/sources/function_vector.cc [hunk 1]
```diff
@@ -1,22 +1,81 @@
...
+
+void FunctionVector::call_all()
+{
+    // Call all functions in the vector.
+    for (const auto& pair : fn_vect_)
+    {
+        const FnType& f = pair.second;
+        f();
+    }
+
```
Comment (TitouanGragnic) [inline Holovibes/sources/function_vector.cc:18]: you can write:

for (const auto& [ _, f]) {
  f();
}

**Review by TitouanGragnic — APPROVED** [review]

**Review by TitouanGragnic — APPROVED** [review]

**Review by Baschtien — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
 ## Changelog
 
+### 14.XX.XX
+
+- Create a second ps1 for registration
+- Enhance the registration process:
+  - The user can choose the radius of the circle where the cross-correlation is computed. A spinbox is added in the GUI.
+  - The reference image is taken from the accumulation queue after the first accumulation at each pipe refresh.
+  - The registration is now working in CLI mode.
```
Comment (Hpn4) [inline CHANGELOG.md:9]: Not related to UI so not necessary to put

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+  - The registration is now in the compute settings. The registration zone (circle radius) is saved in the compute settings.
+- Refactorisation of fn_compute_vect:
+  - Do not inherit std::vector anymore.
+  - Associate each functions to a unique ID. This allows to track each functions and give more flexibility,
+    for example: removing a function from the vector.
```
Comment (Hpn4) [inline CHANGELOG.md:14]: Same not related to UI and user

File: Holovibes/CMakeLists.txt [hunk 4]
```diff
@@ -315,6 +322,8 @@ set(CPACK_NSIS_EXTRA_PREINSTALL_COMMANDS "
 set(SHORTCUT_NAME "$DESKTOP\\\\Holovibes ${PROJECT_VERSION}.lnk")
 set(PSONE_SHORTCUT_NAME "$INSTDIR\\\\scripts\\\\Process Holo Files.lnk")
 set(PSONE_SHORTCUT_SHORTCUT_NAME "$DESKTOP\\\\Process Holo Files ${PROJECT_VERSION}.lnk")
+set(PSONE2_SHORTCUT_NAME "$INSTDIR\\\\scripts\\\\Process Registration.lnk")
```
Comment (GustaveHerve) [inline Holovibes/CMakeLists.txt:325]: todo: change PSONE2 vars names

File: Holovibes/includes/function_vector.hh [hunk 5]
```diff
@@ -1,37 +1,112 @@
-/*! \file
- *
- * \brief #TODO Add a description for this file
- */
```
Comment (Hpn4) [inline Holovibes/includes/function_vector.hh:4]: Why removed ? The \file is used for Doxygen to generate doc for the file

File: Holovibes/includes/API.hh [hunk 6]
```diff
@@ -760,6 +760,11 @@ void display_reticle(bool value);
  */
 void reticle_scale(float value);
 
+/*! \brief Set the new value of the registration zone for the circular mask. Range ]0, 1[.
+ *  \param[in] value The new zone value.
+ */
+void registration_zone(float value);
```
Comment (GustaveHerve) [inline Holovibes/includes/API.hh:766]: maybe rename to set_registration_zone

Comment (GustaveHerve) [inline Holovibes/includes/API.hh:766]: maybe rename not to set_registration_zone but update_registration_zone ? (indicate this is a setter)

File: Holovibes/includes/function_vector.hh [hunk 7]
```diff
@@ -1,37 +1,112 @@
...
+     *  \param[in] remove_condition The condition waited before removing the function from the vector.
+     */
+    void conditionnal_remove(const ushort id, const ConditionType& remove_condition);
+
+    /*! \brief Erase a given function in the vector by its ID.
+     *  Calls `erase` function from `fn_vect_`.
+     *
+     *  \param[in] id The id of the function to remove.
+     */
+    void erase(const ushort id);
```
Comment (Hpn4) [inline Holovibes/includes/function_vector.hh:93]: Maybe put in protected

**Review by GustaveHerve — APPROVED** [review]
top

**Pull request merged**